projects/plugins/jetpack/3rd-party/woocommerce.php

```diff
@@ -31,6 +37,31 @@ function jetpack_woocommerce_integration() {
 	}
 }
 
+/**
+ * Add product post type to sitemap if Woocommerce is present.
+ *
+ * @param array $post_types Array of post types included in sitemap.
+ */
+function jetpack_woocommerce_add_to_sitemap( $post_types ) {
+	$post_types[] = 'product';
+
+	return $post_types;
+}
+
+/**
+ * Skip hidden products when generating the sitemap.
+ *
+ * @param bool    $skip Whether to skip the post.
+ * @param WP_Post $post The post object.
+ */
+function jetpack_woocommerce_skip_hidden_products_in_sitemap( $skip, $post ) {
+	if ( $post !== null && $post->post_type === 'product' ) {
+		$product = wc_get_product( $post->ID );
+		$skip    = ! $product->is_visible();
```

Comment:
I'm waiting for the sitemap generation to happen. Should we be defensive here? `wc_get_product` can return `null` or `false`, in addition to a `WC_Product` object.

Reply:
Thanks. Pushed 4a443f0bd0cb8affa43aa8a0f010daf03359ffe5 addresshing the observation